ipaplatform/base/tasks.py

```diff
@@ -73,6 +73,23 @@ def insert_ca_certs_into_systemwide_ca_store(self, ca_certs):
         Returns True if the operation succeeded, False otherwise.
         """
 
+        try:
```

Comment:
An explanation of the error handling here: before `platform_insert_ca_certs` throws an exception, it logs the context that lets the user figure out exactly what caused the exception to be thrown. That way, the utility functions that manipulate the CA store can exit simply any early when something goes wrong; the logic that handles the failure of those utility functions is in one place (here); and the user gets a useful message such as:

```
ipaplatform.base.tasks: ERROR: Could not populate systemwide CA store
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/ipaplatform/base/tasks.py", line 77, in insert_ca_certs_into_systemwide_ca_store
    if self.platform_insert_ca_certs(ca_certs):
  File "/usr/lib/python3/dist-packages/ipaplatform/debian/tasks.py", line 106, in platform_insert_ca_certs
    self.write_p11kit_certs(paths.IPA_P11_KIT, ca_certs),
  File "/usr/lib/python3/dist-packages/ipaplatform/redhat/tasks.py", line 316, in write_p11kit_certs
    f = open(path, 'w')
FileNotFoundError: [Errno 2] No such file or directory: '/etc/pki/ca-trust/source/ipa.p11-kit'
```
The first log line was produced by the inner utility function before it re-threw the exception.